mitodl/bootcamp-ecommerce
# Change installation and payment deadline logic

#### What are the relevant tickets?
Closes #71

#### What's this PR do?
Changes the payment deadline/installment logic such that we show information about individual installment deadlines rather than just showing the one that is furthest in the future

#### How should this be manually tested?
Check the `/pay` page and select a `Klass` that has multiple installments. Make sure that you see sensible messages when there is a single `Installment` for the `Klass`, and when there are multiple installments with an expired deadline. In the comments I'll link a script that I've used to seed data

#### Where should the reviewer start?
`Payment.js`

#### Any background context you want to provide?
A couple extra bits of work I did in this PR:

1. Changed our APIs to return actual numerical values for `Line.price` and `Installment.amount`, and set our default to avoid coercing `Decimal` values to a string. JSON has support for numerical values, so it didn't make sense to me that we leave it to the front-end to cast those values to numbers.
1. Added an multicolumn index for `Installment` on `klass` and `deadline`. Those values should be unique together and we perform sorting on them, so the index made sense 

#### Screenshots (if appropriate)
![ss 2017-05-16 at 16 03 32](https://cloud.githubusercontent.com/assets/14932219/26126380/596241a0-3a53-11e7-8aac-d47ec14daf6c.png)
![ss 2017-05-16 at 16 03 21](https://cloud.githubusercontent.com/assets/14932219/26126385/5cb49aba-3a53-11e7-8d82-5bba33630b00.png)
![ss 2017-05-16 at 16 03 15](https://cloud.githubusercontent.com/assets/14932219/26126387/5ee42558-3a53-11e7-917e-9ce58365e95d.png)
